src/WebJobs.Script/Host/FunctionMetadataManager.cs

```diff
@@ -18,36 +21,95 @@ namespace Microsoft.Azure.WebJobs.Script
     public class FunctionMetadataManager : IFunctionMetadataManager
     {
         private const string _functionConfigurationErrorMessage = "Unable to determine the primary function script.Make sure atleast one script file is present.Try renaming your entry point script to 'run' or alternatively you can specify the name of the entry point script explicitly by adding a 'scriptFile' property to your function metadata.";
-        private readonly bool _isHttpWorker;
-        private readonly Lazy<ImmutableArray<FunctionMetadata>> _functionMetadataArray;
-        private readonly IOptions<ScriptJobHostOptions> _scriptOptions;
+        private readonly IServiceProvider _serviceProvider;
         private readonly IFunctionMetadataProvider _functionMetadataProvider;
-        private readonly ILogger _logger;
+        private bool _isHttpWorker;
+        private bool _servicesReset = false;
+        private ILogger _logger;
+        private IOptions<ScriptJobHostOptions> _scriptOptions;
+        private ImmutableArray<FunctionMetadata> _functionMetadataFilteredArray;
+        private ImmutableArray<FunctionMetadata> _functionMetadataAllArray;
+        private IEnumerable<IFunctionProvider> _functionProviders;
         private Dictionary<string, ICollection<string>> _functionErrors = new Dictionary<string, ICollection<string>>();
 
-        public FunctionMetadataManager(IOptions<ScriptJobHostOptions> scriptOptions, IFunctionMetadataProvider functionMetadataProvider, IOptions<HttpWorkerOptions> httpWorkerOptions, ILoggerFactory loggerFactory)
+        public FunctionMetadataManager(IOptions<ScriptJobHostOptions> scriptOptions, IFunctionMetadataProvider functionMetadataProvider, IEnumerable<IFunctionProvider> functionProviders, IOptions<HttpWorkerOptions> httpWorkerOptions, IScriptHostManager scriptHostManager, ILoggerFactory loggerFactory)
         {
             _scriptOptions = scriptOptions;
+            _serviceProvider = scriptHostManager as IServiceProvider;
```

Comment:
Can you rename `_serviceProvider` here to something more descriptive like `_jobHostServiceProvider` or something along those lines?